### PR TITLE
Add Bedtools shift

### PR DESCRIPTION
**Pull request by ekushele**
Add module bedtools/shift

## PR checklist

Closes #XXX 

- [v] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [v] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [v] If necessary, include test data in your PR.
- [v] Remove all TODO statements.
- [v] Emit the `versions.yml` file.
- [v] Follow the naming conventions.
- [v] Follow the parameters requirements.
- [v] Follow the input/output options guidelines.
- [v] Add a resource `label`
- [v] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bedtools/shift/main.nf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    label 'process_single'
+
+    conda "bioconda::bedtools=2.30.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bedtools:2.30.0--hc088bd4_0' :
+        'quay.io/biocontainers/bedtools:2.30.0--hc088bd4_0' }"
+
+    input:
+    tuple val(meta), path(bed)
+    path  sizes
```
Comment (nvnieuwk) [inline modules/nf-core/bedtools/shift/main.nf:12]:
```suggestion
    tuple val(meta2), path(sizes)
```
I also think `fai` would be a better input name here

File: modules/nf-core/bedtools/shift/meta.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: bedtools_shift
+description: Shifts each feature by specific number of bases
+keywords:
+  - bed
+  - shiftBed
```
Comment (nvnieuwk) [inline modules/nf-core/bedtools/shift/meta.yml:5]: Can you add some more keywords?

File: modules/nf-core/bedtools/shift/meta.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - bed:
+      type: file
+      description: Input BED file
+      pattern: "*.{bed}"
```
Comment (nvnieuwk) [inline modules/nf-core/bedtools/shift/meta.yml:21]: Can you add the meta2 and the genome file here too?

File: modules/nf-core/fastqc/main.nf [hunk 4]
```diff
@@ -2,10 +2,10 @@ process FASTQC {
     tag "$meta.id"
     label 'process_medium'
 
-    conda "bioconda::fastqc=0.11.9"
+    conda "bioconda::fastqc=0.12.1"
```
Comment (nvnieuwk) [inline modules/nf-core/fastqc/main.nf:5]: Can you please make a separate PR for the changes to `fastqc`?

File: tests/modules/nf-core/bedtools/shift/nextflow.config [hunk 5]
```diff
@@ -0,0 +1,10 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
+    withName: BEDTOOLS_SHIFT {
+        ext.args = '-s 5'
```
Comment (nvnieuwk) [inline tests/modules/nf-core/bedtools/shift/nextflow.config:6]: is `-s` a mandatory argument?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/nf-core/bedtools/shift/main.nf (hunk 1 above)
Comment (ekushele) [reply modules/nf-core/bedtools/shift/main.nf:12]: I kind of copied everything from "bedtools slop" module

File: modules/nf-core/bedtools/shift/meta.yml (hunk 2 above)
Comment (ekushele) [reply modules/nf-core/bedtools/shift/meta.yml:5]: Like what?

File: modules/nf-core/fastqc/main.nf (hunk 4 above)
Comment (ekushele) [reply modules/nf-core/fastqc/main.nf:5]: How can I remove this from the current PR?
I created a PR for my branch "bedtoolsShift" and it was merged into this PR

File: tests/modules/nf-core/bedtools/shift/nextflow.config (hunk 5 above)
Comment (ekushele) [reply tests/modules/nf-core/bedtools/shift/nextflow.config:6]: From bedtools shift: 
`Usage:   bedtools shift [OPTIONS] -i <bed/gff/vcf> -g <genome> [-s <int> or (-p and -m)]`

File: modules/nf-core/bedtools/shift/main.nf (hunk 1 above)
Comment (nvnieuwk) [reply modules/nf-core/bedtools/shift/main.nf]: That module hasn't been updated in a while 😁

Comment (ekushele) [reply modules/nf-core/bedtools/shift/main.nf]: Actually, why "fai" or "sizes" must have meta2? Trying to understand...

File: modules/nf-core/bedtools/shift/meta.yml (hunk 2 above)
Comment (nvnieuwk) [reply modules/nf-core/bedtools/shift/meta.yml:5]: It's mainly because the linter will complain for modules that have so few keywords. A couple suggestions are: regions, fai, genome, base pairs...

File: modules/nf-core/fastqc/main.nf (hunk 4 above)
Comment (nvnieuwk) [reply modules/nf-core/fastqc/main.nf:5]: You can undo those commits normally :) Otherwise you can just copy the code from `master` into those files and it should be fixed too

File: tests/modules/nf-core/bedtools/shift/nextflow.config (hunk 5 above)
Comment (nvnieuwk) [reply tests/modules/nf-core/bedtools/shift/nextflow.config:6]: Allright it's fine as it is now :)

File: modules/nf-core/bedtools/shift/meta.yml (hunk 3 above)
Comment (ekushele) [reply modules/nf-core/bedtools/shift/meta.yml:26]: I can't find the other converstaion, but why is there a need for "meta2" if it's only a file of chr sizes?

Comment (nvnieuwk) [reply modules/nf-core/bedtools/shift/meta.yml:26]: Some pipelines use multiple kinds of genomes and need a way to parse this information. This can be done through a meta field, so in this case it makes sense to add it :)

File: modules/nf-core/fastqc/main.nf (hunk 4 above)
Comment (ekushele) [reply modules/nf-core/fastqc/main.nf]: done

File: tests/config/pytest_modules.yml [hunk 6]
```diff
@@ -353,6 +353,10 @@ bedtools/multiinter:
   - modules/nf-core/bedtools/multiinter/**
   - tests/modules/nf-core/bedtools/multiinter/**
 
+bedtools/slop:
```
Comment (nvnieuwk) [inline tests/config/pytest_modules.yml:356]:
```suggestion
bedtools/shift:
```

Comment (ekushele) [reply tests/config/pytest_modules.yml]: sorry...!

Comment (nvnieuwk) [reply tests/config/pytest_modules.yml]: No problem! (you wouldn't want to know how many times they had to fix copy paste errors from me 😁)

File: modules/nf-core/bedtools/shift/meta.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  - chrom_sizes:
+      type: file
+      description: Chromosome sizes file
+      pattern: "*{.sizes,.txt}"
```
Comment (nvnieuwk) [inline modules/nf-core/bedtools/shift/meta.yml:30]:
```suggestion
      pattern: "*{.sizes,.txt,.fai}"
```

**Review by nvnieuwk — APPROVED** [review]
One small comment and then you are good to go 🥳

**Comment by ekushele** [comment]
@nvnieuwk Thank you very much!
